src/ImageSharp/Formats/Gif/GifEncoderCore.cs

```diff
@@ -141,7 +141,7 @@ public void Encode<TPixel>(Image<TPixel> image, Stream stream, CancellationToken
     private void EncodeFrames<TPixel>(
         Stream stream,
         Image<TPixel> image,
-        IndexedImageFrame<TPixel> quantized,
+        IndexedImageFrame<TPixel>? quantized,
```

Comment:
I haven't pulled this down to check locally but I think quantized is never null when calling this method. We set it to null but I don't think we need to.

Reply:
I also thought that. I changed it and the tests failed.

It can be null because the method is calling itself. The problem is, that quantized is passed from the outside for the first frame. For the other frames it is created.

This could be probably get a refactoring but probably way out of scope of this change

Reply:
`EncodeFrames` calls `EncodeFrame` - Not the most penetrable code I know but I actually refactored it to make it more readable recently! 🤣

I managed to tweak it to work without nullifying.

Reply:
I thought I tried the exact same thing but got test failures afterwards. But could also been in another file. Not so sure anymore touched too much lately to remember^^

Reply:
I've been thoroughly impressed with how well you've navigated the codebase. There's a LOT to take in!